pkgs/by-name/se/sequoia-wot/package.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitLab,
+  libclang,
+  pkg-config,
+  nettle,
+  openssl,
+  sqlite,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "sequoia-wot";
+  version = "0.11.0";
+
+  src = fetchFromGitLab {
+    owner = "sequoia-pgp";
+    repo = "sequoia-wot";
+    rev = "v${version}";
+    hash = "sha256-qSf2uESsMGUEvAiRefpwxHKyizbq5Sst3SpjKaMIWTQ=";
+  };
+
+  cargoPatches = [
+    ./01-cargo-lock.patch
+    ./02-cargo-toml.patch
+  ];
+
+  cargoHash = "sha256-ZPVvRulCXG81eTuL2oettUrrFWEmZk8VzD9YB+P+wVc=";
+
+  nativeBuildInputs = [
+    pkg-config
+    rustPlatform.bindgenHook
```

Comment:
```suggestion
    rustPlatform.bindgenHook
    installShellFiles
```

Locally I noticed these log messages during the build:

```
writing man page to /build/source/target/x86_64-unknown-linux-gnu/release/build/squoia-wot-fb11ecd97b1459ce/out/sq-wot.1
writing man page to /build/source/target/x86_64-unknown-linux-gnu/release/build/squoia-wot-fb11ecd97b1459ce/out/sq-wot-authenticate.1
writing man page to /build/source/target/x86_64-unknown-linux-gnu/release/build/squoia-wot-fb11ecd97b1459ce/out/sq-wot-lookup.1
writing man page to /build/source/target/x86_64-unknown-linux-gnu/release/build/squoia-wot-fb11ecd97b1459ce/out/sq-wot-identify.1
writing man page to /build/source/target/x86_64-unknown-linux-gnu/release/build/squoia-wot-fb11ecd97b1459ce/out/sq-wot-list.1
writing man page to /build/source/target/x86_64-unknown-linux-gnu/release/build/squoia-wot-fb11ecd97b1459ce/out/sq-wot-path.1
completion file is generated: "/build/source/target/x86_64-unknown-linux-gnu/relese/build/sequoia-wot-fb11ecd97b1459ce/out/sq-wot.bash"
completion file is generated: "/build/source/target/x86_64-unknown-linux-gnu/relese/build/sequoia-wot-fb11ecd97b1459ce/out/sq-wot.fish"
completion file is generated: "/build/source/target/x86_64-unknown-linux-gnu/relese/build/sequoia-wot-fb11ecd97b1459ce/out/_sq-wot"
completion file is generated: "/build/source/target/x86_64-unknown-linux-gnu/relese/build/sequoia-wot-fb11ecd97b1459ce/out/_sq-wot.ps1"
completion file is generated: "/build/source/target/x86_64-unknown-linux-gnu/relese/build/sequoia-wot-fb11ecd97b1459ce/out/sq-wot.elv"
```

Could you find a way to detect the paths to these files and install these? Maybe something like this (used in `sequoia-sq`) will work:

```nix
  env.ASSET_OUT_DIR = "/tmp/";

  postInstall = ''
    installManPage /tmp/man-pages/*.*
    installShellCompletion \
      --cmd sq \
      --bash /tmp/shell-completions/sq.bash \
      --fish /tmp/shell-completions/sq.fish \
      --zsh /tmp/shell-completions/_sq
  '';
```

Reply:
The output path like `build/squoia-wot-fb11ecd97b1459ce/out` is random. To search these outputs in `postInstall` may be very dirty.

There is an issue about gettin the output path: https://github.com/rust-lang/cargo/issues/9661

I also opened an issue for asking upstream to support generate these files runtime: https://gitlab.com/sequoia-pgp/sequoia-wot/-/issues/56

Reply:
The output path in sequoia-wot is `OUT_DIR` in https://gitlab.com/sequoia-pgp/sequoia-wot/-/blob/v0.11.0/build.rs?ref_type=tags#L99 , not `ASSET_OUT_DIR`. It's a cargo defined variable.

Reply:
> The output path in sequoia-wot is `OUT_DIR` in [gitlab.com/sequoia-pgp/sequoia-wot/-/blob/v0.11.0/build.rs?ref_type=tags#L99](https://gitlab.com/sequoia-pgp/sequoia-wot/-/blob/v0.11.0/build.rs?ref_type=tags#L99) , not `ASSET_OUT_DIR`. It's a cargo defined variable.

So did you manage to use this env var to find the man pages and shell completion files?

Reply:
> > The output path in sequoia-wot is `OUT_DIR` in [gitlab.com/sequoia-pgp/sequoia-wot/-/blob/v0.11.0/build.rs?ref_type=tags#L99](https://gitlab.com/sequoia-pgp/sequoia-wot/-/blob/v0.11.0/build.rs?ref_type=tags#L99) , not `ASSET_OUT_DIR`. It's a cargo defined variable.
> 
> So did you manage to use this env var to find the man pages and shell completion files?

cargo only set it during building, we can't get it. Upstream https://github.com/rust-lang/cargo/issues/9661 has discussed it a lot. It's now not a reasonable solution for me.

Reply:
> > So did you manage to use this env var to find the man pages and shell completion files?
> 
> cargo only set it during building, we can't get it. Upstream [rust-lang/cargo#9661](https://github.com/rust-lang/cargo/issues/9661) has discussed it a lot. It's now not a reasonable solution for me.

Oh I understand. I thought you meant to say that upstream's `build.rs` file indicates we can use `$OUT_DIR` instead of `$ASSET_OUT_DIR`. In any case, all of the `sequoia-*` packages are influenced by similar issues, and the way we install these shell files is inconsistent through out all of them. I tested the following to work:

```diff
diff --git i/pkgs/by-name/se/sequoia-wot/package.nix w/pkgs/by-name/se/sequoia-wot/package.nix
index dd12f990049a..4b5c6e27ea8e 100644
--- i/pkgs/by-name/se/sequoia-wot/package.nix
+++ w/pkgs/by-name/se/sequoia-wot/package.nix
@@ -2,6 +2,7 @@
   lib,
   rustPlatform,
   fetchFromGitLab,
+  installShellFiles,
   libclang,
   pkg-config,
   nettle,
@@ -30,6 +31,7 @@ rustPlatform.buildRustPackage rec {
   nativeBuildInputs = [
     pkg-config
     rustPlatform.bindgenHook
+    installShellFiles
   ];
 
   buildInputs = [
@@ -38,10 +40,31 @@ rustPlatform.buildRustPackage rec {
     openssl
     sqlite
   ];
+  # Install shell completion files and manual pages. Unfortunatly it is hard to
+  # predict the paths to all of these files generated during the build, and it
+  # is impossible to control these using `$OUT_DIR` or alike, as implied by
+  # upstream's `build.rs`. This is a general Rust issue also discussed in
+  # https://github.com/rust-lang/cargo/issues/9661, also discussed upstream at:
+  # https://gitlab.com/sequoia-pgp/sequoia-wot/-/issues/56
+  postInstall = ''
+    installShellCompletion --cmd sq-wot \
+      --bash target/*/release/build/sequoia-wot-*/out/sq-wot.bash \
+      --fish target/*/release/build/sequoia-wot-*/out/sq-wot.fish \
+      --zsh  target/*/release/build/sequoia-wot-*/out/_sq-wot
+    # Also elv and powershell are generated there
+    installManPage \
+      target/*/release/build/sequoia-wot-*/out/sq-wot.1 \
+      target/*/release/build/sequoia-wot-*/out/sq-wot-authenticate.1 \
+      target/*/release/build/sequoia-wot-*/out/sq-wot-lookup.1 \
+      target/*/release/build/sequoia-wot-*/out/sq-wot-identify.1 \
+      target/*/release/build/sequoia-wot-*/out/sq-wot-list.1 \
+      target/*/release/build/sequoia-wot-*/out/sq-wot-path.1
+  '';
 
   buildFeatures = [ "sequoia-openpgp/default" ];
 
   nativeCheckInputs = [ gnupg ];
+  doCheck = true;
 
   meta = with lib; {
     description = "A Rust CLI tool for authenticating bindings and exploring a web of trust";
```

Which is based upon what's in `sequoia-sqop`.

Reply:
done